client/my-sites/site-settings/seo-settings/form.jsx

```diff
@@ -313,6 +315,14 @@ export const SeoForm = React.createClass( {
 			updatedOptions.advanced_seo_front_page_description = this.state.frontPageMetaDescription;
 		}
 
+		// Empty arrays for title formats were not working properly with saveSiteSettings.
+		// We are replacing them with empty strings here to allow users to delete custom title formats.
+		const prepareEmptyFormats = formats =>
+			mapValues( formats, format => isArray( format ) && 0 === format.length ? '' : format );
+
+		// Replace empty arrays with empty strings to fix custom format deletion bug.
+		updatedOptions.advanced_seo_title_formats = prepareEmptyFormats( updatedOptions.advanced_seo_title_formats );
```

Comment:
note that we don't have much reason to split the function call from its definition since it's just a single line. we could inline here

```js
updatedOptions.advanced_seo_title_formats = mapValues(
	updatedOptions.advanced_seo_title_formats,
	format => isArray( format ) && 0 === format.length ? '' : format,
);
```